version: "3"
volumes:
  dbbackups:
  postgis-data:
  session-redis:
  access-redis:
services:
  main:
    image: patreon
    expose:
      - "80"
      - "443"
    ports:
      - "443:443"
      - "80:80"
    networks:
      - default
    volumes:
      - ./logs:/app/logs
      - ./media:/app/media
      - ./patreon-secrt:/app/patreon-secrt
    depends_on:
      - session-redis
      - access-redis
      - patreon-bd
  session-redis:
    image: "redis:alpine"
    expose:
      - "6379"
    ports:
      - "6379:6379"
    volumes:
      - session-redis:/data
  access-redis:
    image: "redis:alpine"
    expose:
      - "6380"
    ports:
      - "6380:6380"
    volumes:
      - access-redis:/data
  patreon-bd:
    image: postgres:latest
    expose:
      - "5432"
    volumes:
      - ./scripts/user.sql:/docker-entrypoint-initdb.d/1_init.sql
      - ./scripts/start_initialization.sql:/docker-entrypoint-initdb.d/2_init.sql
      - postgis-data:/var/lib/postgresql
    environment:
      - POSTGRES_PASSWORD=pyaterochka
      - POSTGRES_USER=patreon
      - POSTGRES_DB=restapi_dev
    ports:
      - "5432:5432"
    restart: on-failure
    healthcheck:
      test: "exit 0"
    deploy:
      resources:
        limits:
          cpus: '2'
          memory: 4G
#  pgadmin:
#    container_name: pgadmin_container
#    image: dpage/pgadmin4:5.7
#    environment:
#      PGADMIN_DEFAULT_EMAIL: "m@mail.ru"
#      PGADMIN_DEFAULT_PASSWORD: "pyaterochka"
#      PGADMIN_CONFIG_SERVER_MODE: "False"
#    volumes:
#      - ./pgadmin:/var/lib/pgadmin
#    ports:
#      - "5050:80"
#    restart: unless-stopped
#    deploy:
#      resources:
#        limits:
#          cpus: '0.5'
#          memory: 1G
#    networks:
#      - postgres
networks:
  postgres:
    driver: bridge
